evm/src/executor/backend/mod.rs

```diff
@@ -1281,6 +1338,13 @@ impl Default for BackendInner {
             next_fork_id: Default::default(),
             persistent_accounts: Default::default(),
             precompile_id: revm::precompiles::SpecId::LATEST,
+            // grant the cheatcode,default test and caller address access to execute cheatcodes
+            // itself
+            cheatcode_access_accounts: HashSet::from([
+                CHEATCODE_ADDRESS,
+                TEST_CONTRACT_ADDRESS,
```

Comment:
I know invariant does this too, but really we don't know what the test contract address is before runtime, so we can't rely on this constant. would add a note at least so we can quickly find it/fix it later

Reply:
right, since we also inject new deployed addresses, adding `TEST_CONTRACT_ADDRESS` should be redundant here, but I don't think it hurts including this by default